app/components/osf-navbar/component.ts

```diff
@@ -0,0 +1,80 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import config from 'ember-get-config';
+import { osfServices } from 'ember-osf-web/const/service-links';
+import AnalyticsMixin from 'ember-osf-web/mixins/analytics';
+
+const HOME_APP = 'HOME';
+
+/**
+ * Display the OSF navbar
+ *
+ * @class osf-navbar
+ */
+export default class OsfNavbar extends Component.extend(AnalyticsMixin) {
+    /**
+     * Action run when the user clicks "Sign In"
+     *
+     * @property loginAction
+     * @type {Action}
+     */
+    loginAction: () => void;
+
+    /**
+     * The URL to use for signup
+     * May be overridden, e.g. for special campaign pages
+     *
+     * @property signupUrl
+     * @type {String}
+     */
+    signupUrl: string = this.signupUrl || `${config.OSF.url}register`;
+
+    /**
+     * The URL to redirect to after logout
+     *
+     * @property redirectUrl
+     * @type {String}
+     */
+    redirectUrl: string = this.redirectUrl || '';
+
+    // Private properties
+    session = service('session');
+
+    // TODO: When used in other apps, update to expect these as arguments or from the config
+    hostAppName: string = HOME_APP;
+    linksComponent: string = 'osf-navbar/home-links';
```

Comment:
I like this approach *much* better than the one "link builder for all the apps" helper.